components/brave_private_cdn/private_cdn_request_helper.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_private_cdn/private_cdn_request_helper.h"
+
+#include <utility>
+
+#include "brave/components/brave_private_cdn/headers.h"
+#include "net/base/load_flags.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/mojom/url_response_head.mojom.h"
+
+namespace brave_private_cdn {
+
+namespace {
+
+const unsigned int kRetriesCountOnNetworkChange = 3;
+
+}  // namespace
+
+PrivateCDNRequestHelper::PrivateCDNRequestHelper(
+    net::NetworkTrafficAnnotationTag annotation_tag,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : annotation_tag_(annotation_tag),
+      url_loader_factory_(url_loader_factory) {}
+
+PrivateCDNRequestHelper::~PrivateCDNRequestHelper() {}
+
+void PrivateCDNRequestHelper::DownloadToString(
+    const GURL& url,
+    DownloadToStringCallback callback,
+    bool auto_retry_on_network_change /* = true */,
+    size_t max_body_size /* = 5 * 1024 * 1024 */) {
+  // Do not allow unbounded max body
+  DCHECK(max_body_size != = -1u);
+  // Create private request
+  auto request = std::make_unique<network::ResourceRequest>();
+  request->url = url;
+  // Load flags allow cache, although we may want to accept a param for this,
+  // if we are requesting static urls which do not support etag.
+  // Don't send or save cookies
+  request->load_flags = net::LOAD_DO_NOT_SAVE_COOKIES;
+  request->credentials_mode = network::mojom::CredentialsMode::kOmit;
+  request->method = net::HttpRequestHeaders::kGetMethod;
+  // Don't send any identifying information, such as language or user agent
+  for (auto entry : brave::private_cdn_headers)
```

Comment:
`brave::private_cdn_headers` should be a `base::NoDestructor` instance accessible (and constructible) via some getter like `brave::GetPrivateCDNHeaders()`.

Reply:
Would prefer to change in a follow-up. This is not introduced in this PR, and would prefer to limit this PR to fix the regression and not refactor